frontend/src/App.tsx

```diff
@@ -16,8 +17,32 @@ import OpenInNewIcon from "@mui/icons-material/OpenInNew";
 import Typography from "@mui/joy/Typography";
 import { BrowserRouter, Routes, Route, Link } from "react-router-dom";
 import { HEADER_HEIGHT_PX, HEADER_MARGIN_PX } from "./styles/sizes";
+import { Year, LeaderboardEntry } from "./types";
+import { DEFAULT_YEAR } from "./constants";
 
 export default function App() {
+    const [leaderboard, setLeaderboard] = useState<LeaderboardEntry[]>([]);
+    const [leaderboardYear, setLeaderboardYear] = useState<Year>(DEFAULT_YEAR);
+    const [loadingLeaderboard, setLoadingLeaderboard] = useState(false);
+
+    const getLeaderboard = () => {
+        setLoadingLeaderboard(true);
+
+        axios
+            .get("https://api.waroftheringcommunity.net:8080/leaderboard", {
+                params: { year: leaderboardYear },
+            })
+            .then((response) => {
+                setLeaderboard(response.data.entries as LeaderboardEntry[]);
```

Comment:
We should absolutely look into https://docs.servant.dev/en/stable/tutorial/Javascript.html particularly https://docs.servant.dev/en/stable/tutorial/Javascript.html#axios-support

Would be great to not risk breaking this every/any time the API changes.

Reply:
This would be next-level 💯 Added an issue for it so I don't forget! As a stopgap I should really add types for server responses to make it more noticeable when frontend and backend types get out of sync. Having to dig through the code for axios calls is not great

Reply:
Agreed - but maybe it's not worth doing that until we are auto-generating the calls anyway? This could be something we investigate together.

Reply:
Sweet won't memorialize it in my sticky note then, sounds like a fun project